cmd/bridge/config/auth/authoptions.go

```diff
@@ -0,0 +1,281 @@
+package auth
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+
+	"github.com/openshift/console/pkg/auth"
+	"github.com/openshift/console/pkg/flags"
+	"github.com/openshift/console/pkg/proxy"
+	"github.com/openshift/console/pkg/server"
+	"github.com/openshift/console/pkg/serverconfig"
+)
+
+type AuthOptions struct {
+	AuthType string
+
+	IssuerURL            string
+	ClientID             string
+	ClientSecret         string
+	ClientSecretFilePath string
+	CAFilePath           string
+
+	InactivityTimeoutSeconds int
+	LogoutRedirect           string
+}
+
+type CompletedOptions struct {
+	*completedOptions
+}
+
+type completedOptions struct {
+	AuthType string
+
+	IssuerURL    *url.URL
+	ClientID     string
+	ClientSecret string
+	CAFilePath   string
+
+	InactivityTimeoutSeconds int
+	LogoutRedirectURL        *url.URL
+}
+
+func NewAuthOptions() *AuthOptions {
+	return &AuthOptions{}
```

Comment:
set default values here instead of in 'Complete'?

Reply:
We can only do that after `ApplyConfig()` is called